api/oneclick.py

```diff
@@ -343,14 +345,35 @@ def update_licensepool_for_identifier(self, isbn, availability):
         if not availability:
             licenses_available = 0
         licenses_owned = licenses_available
+
+
+        # If possible, create a FormatData object representing
+        # how the book is available.
+        formats = []
+
+        # Note that these strings are different from the similar strings
+        # found in "fileFormat" when looking at a patron's loans.
+        # "ebook" (a medium) versus "EPUB" (a format). Unfortunately we
+        # 
+        delivery_type = None
+        medium = medium.lower()
+        if medium == 'ebook':
+            delivery_type = Representation.EPUB_MEDIA_TYPE
+        elif medium == 'eaudio':
+            # A questionable assumption.
+            delivery_type = Representation.MP3_MEDIA_TYPE
+
+        if delivery_type:
+            formats.append(FormatData(delivery_type, None))
```

Comment:
I wiped the database and tried again from scratch, and this time the books were created with the proper delivery mechanisms (the ones defined in core). So this might have been fixed by some change I made later in this branch.